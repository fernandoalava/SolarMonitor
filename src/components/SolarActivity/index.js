import React from "react";
import PropTypes from "prop-types";
import moment from "moment";
import CircularProgress from "@material-ui/core/CircularProgress";

import {
  LineChart,
  Line,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  Legend
} from "recharts";

export const SolarActivity = props => {
  const { activity } = props;
  const now = moment();
  const next24hours = moment().add(24, "hours");

  const data = props.activity
    .filter(item => moment(item.axes.time).isBetween(now, next24hours))
    .map(item => ({
      time: moment(item.axes.time).format("HH:mm"),
      watts: Math.floor(item.data.av_swsfcdown)
    }));

  return activity.length >= 1 ? (
    <LineChart
      width={500}
      height={150}
      data={data}
      margin={{
        top: 5,
        right: 30,
        left: 20,
        bottom: 5
      }}
    >
      <CartesianGrid strokeDasharray="3 3" />
      <XAxis dataKey="time" />
      <YAxis dataKey="watts" />
      <Tooltip />
      <Legend />
      <Line
        type="monotone"
        dataKey="watts"
        stroke="#8884d8"
        activeDot={{ r: 8 }}
        name="Watts m-2"
      />
    </LineChart>
  ) : (
    <CircularProgress />
  );
};

SolarActivity.propTypes = {
  activity: PropTypes.array.isRequired
};

SolarActivity.defaultProps = {
  activity: []
};

export default SolarActivity;
