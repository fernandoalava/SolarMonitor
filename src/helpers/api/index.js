import config from "../../config";
import axios from "axios";

export const fetchCloudData = async (dataHandler, errorHanlder) => {
  try {
    let response = await axios.get(
      `${config.protocol}://${
        config.planetOSDomain
      }/v1/datasets/bom_access-g_global_40km/point?origin=dataset-details&apikey=${
        config.apiKey
      }&count=50&lat=49.6&lon=-50.5&var=av_ttl_cld`
    );
    dataHandler(response.data.entries);
  } catch (err) {
    errorHanlder(err);
  }
};

export const fetchSolarData = async (dataHandler, errorHanlder) => {
  try {
    let response = await axios.get(
      `${config.protocol}://${
        config.planetOSDomain
      }/v1/datasets/bom_access-g_global_40km/point?origin=dataset-details&apikey=${
        config.apiKey
      }&count=50&lat=49.6&lon=-50.5&var=av_swsfcdown`
    );
    dataHandler(response.data.entries);
  } catch (err) {
    errorHanlder(err);
  }
};
