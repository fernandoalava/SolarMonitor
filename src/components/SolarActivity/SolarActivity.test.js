import { shallow } from "enzyme";
import * as React from "react";
import SolarActivity from ".";
import moment from "moment";
import { solarActivityDataset } from "../../helpers/index";

describe("SolarActivity", () => {
  it("Renders correctly without data", () => {
    const wrapper = shallow(<SolarActivity activity={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders correctly with data", () => {
    const wrapper = shallow(
      <SolarActivity activity={solarActivityDataset.entries} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
