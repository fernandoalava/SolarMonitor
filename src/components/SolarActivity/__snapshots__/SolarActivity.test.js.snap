// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SolarActivity Renders correctly with data 1`] = `
<LineChart
  barCategoryGap="10%"
  barGap={4}
  data={
    Array [
      Object {
        "time": "18:00",
        "watts": 724,
      },
      Object {
        "time": "21:00",
        "watts": 352,
      },
      Object {
        "time": "00:00",
        "watts": 46,
      },
      Object {
        "time": "03:00",
        "watts": 0,
      },
      Object {
        "time": "06:00",
        "watts": 0,
      },
      Object {
        "time": "09:00",
        "watts": 17,
      },
      Object {
        "time": "12:00",
        "watts": 224,
      },
      Object {
        "time": "15:00",
        "watts": 397,
      },
    ]
  }
  height={150}
  layout="horizontal"
  margin={
    Object {
      "bottom": 5,
      "left": 20,
      "right": 30,
      "top": 5,
    }
  }
  reverseStackOrder={false}
  stackOffset="none"
  width={500}
>
  <CartesianGrid
    fill="none"
    horizontal={true}
    horizontalFill={Array []}
    horizontalPoints={Array []}
    stroke="#ccc"
    strokeDasharray="3 3"
    vertical={true}
    verticalFill={Array []}
    verticalPoints={Array []}
  />
  <XAxis
    allowDataOverflow={false}
    allowDecimals={true}
    allowDuplicatedCategory={true}
    dataKey="time"
    domain={
      Array [
        0,
        "auto",
      ]
    }
    height={30}
    hide={false}
    mirror={false}
    orientation="bottom"
    padding={
      Object {
        "left": 0,
        "right": 0,
      }
    }
    reversed={false}
    scale="auto"
    tickCount={5}
    type="category"
    width={0}
    xAxisId={0}
  />
  <YAxis
    allowDataOverflow={false}
    allowDecimals={true}
    allowDuplicatedCategory={true}
    dataKey="watts"
    domain={
      Array [
        0,
        "auto",
      ]
    }
    height={0}
    hide={false}
    mirror={false}
    orientation="left"
    padding={
      Object {
        "bottom": 0,
        "top": 0,
      }
    }
    reversed={false}
    scale="auto"
    tickCount={5}
    type="number"
    width={60}
    yAxisId={0}
  />
  <Tooltip
    active={false}
    animationDuration={400}
    animationEasing="ease"
    contentStyle={Object {}}
    coordinate={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    cursor={true}
    cursorStyle={Object {}}
    filterNull={true}
    isAnimationActive={true}
    itemStyle={Object {}}
    labelStyle={Object {}}
    offset={10}
    separator=" : "
    useTranslate3d={false}
    viewBox={
      Object {
        "x1": 0,
        "x2": 0,
        "y1": 0,
        "y2": 0,
      }
    }
    wrapperStyle={Object {}}
  />
  <Legend
    align="center"
    iconSize={14}
    layout="horizontal"
    verticalAlign="bottom"
  />
  <Line
    activeDot={
      Object {
        "r": 8,
      }
    }
    animateNewValues={true}
    animationBegin={0}
    animationDuration={1500}
    animationEasing="ease"
    connectNulls={false}
    dataKey="watts"
    dot={true}
    fill="#fff"
    hide={false}
    isAnimationActive={true}
    legendType="line"
    name="Watts m-2"
    onAnimationEnd={[Function]}
    onAnimationStart={[Function]}
    points={Array []}
    stroke="#8884d8"
    strokeWidth={1}
    type="monotone"
    xAxisId={0}
    yAxisId={0}
  />
</LineChart>
`;

exports[`SolarActivity Renders correctly without data 1`] = `<WithStyles(ForwardRef(CircularProgress)) />`;
